dace/transformation/interstate/fpga_global_to_local.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
+""" Contains a transformation that changes the storage type of a Global FPGA variable to Local when certains
+    conditions are met.
+"""
+
+import networkx as nx
+
+from dace import registry, properties
+from dace.transformation import transformation
+from dace import symbolic, dtypes
+from dace.sdfg import nodes, trace_nested_access
+from dace import config
+
+
+@registry.autoregister
+@properties.make_properties
+class FPGAGlobalToLocal(transformation.Transformation):
+    """ Implements the FPGAGlobalToLoca transformation, which takes an entire
+        SDFG  and changes the storage type of a Global FPGA data container to Local in the following situation:
+        - data the is transient,
+        - and the data is not a transient shared with other states,
+        - and data has a compile-time known size. """
+    @staticmethod
+    def annotates_memlets():
+        return True
+
+    @staticmethod
+    def expressions():
+        # Match anything
+        return [nx.DiGraph()]
+
+    @staticmethod
+    def can_be_applied(graph, candidate, expr_index, sdfg, strict=False):
+        # It can always be applied
+        return True
```

Comment:
Why are you not checking if it can be applied? If you don't want to repeat code, then you can make a helper function. I don't like having a transformation that can be "applied" but doesn't do anything

Reply:
Should this return True if at least one array can be converted to Local?

Reply:
Ideally it should match on a particular array that could be converted to local -- although maybe the best we can do is match a data node pointing to this array (@tbennun any better idea?), so you get N matches for N arrays that could be converted from glboal to local